### PR TITLE
MINOR: Add tests for some search-form fixes

### PR DESCRIPTION
**Pull request by sminnee**
These 2 tests validate that the following bugs have been fixed.

https://github.com/silverstripe/silverstripe-framework/issues/3212
https://github.com/silverstripe/silverstripe-cms/issues/1452

### CODE REVIEW

File: tests/php/Search/DatabaseSearchEngineTest.php [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    {
+        $page = new SiteTree();
+        $page->Title = "Unrelated page";
+        $page->write();
+        $page->doPublish();
+
+        $results = DB::get_conn()->searchEngine([ SiteTree::class, File::class ], "foo>*", 0, 100, "Relevance DESC", "", true);
+
+        // We're not trying to match this query, just confirm that it successfully executes
+        $this->assertEquals(0, $results->count());
```
Comment (robbieaverill) [inline tests/php/Search/DatabaseSearchEngineTest.php]: Super minor, but you can use assertCount() here:

```suggestion
        $this->assertCount(0, $results);
```

**Comment by robbieaverill** [comment]
PG tests aren't passing =/

File: tests/php/Search/DatabaseSearchEngineTest.php (hunk 1 above)
Comment (sminnee) [reply tests/php/Search/DatabaseSearchEngineTest.php]: Sweet I'll add that, and in the other place, and quote Relevance to fix PG.

**Comment by ScopeyNZ** [comment]
Mmmn TDD 😉

**Pull request merged**